workers/unity/Packages/com.improbable.gdk.querybasedinterest/Interest.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    public class Interest
+    {
+        internal ComponentInterest.Query query;
+
+        public static implicit operator ComponentInterest.Query(Interest interest)
+        {
+            return interest.query;
+        }
+        
+        protected internal Interest()
+        {
+            query = new ComponentInterest.Query
+            {
+                FullSnapshotResult = true
+            }; 
+        }
+        
+        public static Interest Query(ComponentInterest.QueryConstraint constraint)
+        {
+            var interest = new Interest();
+            interest.query.Constraint = constraint;
+            return interest;
+        }
+        
+        public Interest Frequency(float frequency)
+        {
+            query.Frequency = frequency;
+            return this;
+        }
+        
+        public Interest Result<T>() where T : ISpatialComponentData
```

Comment:
Yeah I thought the same after writing it but left it in to hear other thoughts. Originally it was `params uint[] resultIds` without the generic - probably will revert to that.

Reply:
The problem with params is that there is then an allocation in the conversion to List. However providing we are caching the results at startup I don't think it's that bad.

Reply:
How about a method like 
```
.WithResult<MyFirstComponent>()
.WithResult<MySecondComponent>()
.WithResult<MyThirdComponent>()
```
?

Reply:
Could do and it would be more general than what is there now. It's quite verbose though and it means you can't just save an array of components that different queries might need, if multiple queries need the same components. You could add some other type in to save a list created like that but that seems like adding epicycles.

Reply:
the current implementation already lets you stack up the `.Result` bit as many times as you want so that's not an issue